core/trino-spi/src/main/java/io/trino/spi/block/SingleRowBlockWriter.java

```diff
@@ -62,6 +57,13 @@ public BlockBuilder getFieldBlockBuilder(int fieldIndex)
         return fieldBlockBuilders[fieldIndex];
     }
 
+    public void reset(int rowIndex)
```

Comment:
there should really be 2 methods. This one that sets `this.rowIndex` to `-1`
and another one that just sets `this.rowIndex` to non -1 value. The latter one doesn't have to set `currentFieldIndexToWrite` and `fieldBlockBuilderReturned`